sdk/src/main/java/io/customer/sdk/data/store/Client.kt

```diff
@@ -10,24 +10,74 @@ import java.util.*
  * cannot be change (e.g. the SDK version)
  *
  * @property source name of the client source to append with user-agent
- * @property identifier name only used to identify client, can be anything;
- * default value is lowercase of the name
+ * @property sdkVersion version of the SDK used
  */
-sealed class Client(
+class Client private constructor(
     val source: String,
-    private val identifier: String? = source.lowercase(Locale.ENGLISH)
+    val sdkVersion: String
 ) {
-    object Android : Client(source = "Android")
-    object ReactNative : Client(source = "ReactNative")
-    object Expo : Client(source = "Expo")
-    class Other(value: String) : Client(source = value, identifier = null)
+    override fun toString(): String = "$source Client/$sdkVersion"
 
     companion object {
-        fun fromSource(source: String): Client {
-            val identifier = source.lowercase(Locale.ENGLISH)
-            return listOf(Android, ReactNative, Expo).find { client ->
-                client.identifier == identifier
-            } ?: Other(value = source)
+        @VisibleForTesting
+        internal const val SOURCE_ANDROID = "Android"
+
+        @VisibleForTesting
+        internal const val SOURCE_EXPO = "Expo"
+
+        @VisibleForTesting
+        internal const val SOURCE_REACT_NATIVE = "ReactNative"
+
+        /**
+         * Creates the right client based on name provided (case insensitive).
+         * <p/>
+         * Use this if you don't want to use direct method or want to use same
+         * method from different SDK wrappers.
+         *
+         * @param source name of the source platform.
+         * @param sdkVersion version of the SDK being used.
+         * @return new instance of client against the provided data.
+         */
+        fun create(source: String, sdkVersion: String): Client = when {
```

Comment:
I would recommend doing something like this, because there are chances of string mismatch in this case. 

```
sealed class Client(source: String, sdkVerions: String) {
data class Android(version) : Client(android, version)
data class React(version) : Client(react , version)
data class Other(source : String, version : String) : Client(source , version)
}
```

Reply:
Yeah, this makes sense. I made it little complex for some unknown reason I don't remember 😅 Updated it to use sealed classes 👨🏻‍💻